build/action.yml

```diff
@@ -0,0 +1,49 @@
+name: Matter Snap Builder
+description: |
+  This action clones the source code, builds the snap using and 
+  uploads the resulting snap as a build artifact
+
+inputs:
+  repo:
+    description: Source code repo with snap
+    required: false
+    default: ${{github.repository}}
+  branch:
+    description: Source code repo branch
+    required: false
+    default: ${{github.ref}}
+
+outputs:
+  snap:
+    description: Relative path to built snap
+    value: ${{steps.build.outputs.snap}}
+
+runs:
+  using: composite
+
+  steps:
+    # Clone the source project
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        repository: ${{inputs.repo}}
+        ref: ${{inputs.branch}}
+
+    # Get snap base version
+    - shell: bash
+      run: |
+        echo "base=$(sed -n 's/^base: //p' snap/snapcraft.yaml)" >> $GITHUB_ENV
+
+    # Build the snap
+    - uses: snapcore/action-build@v1
+      id: build
+      with:
+        snapcraft-args: ${{ env.base == 'core22' && '--verbose' || '' }}
```

Comment:
Checking the base to toggle verbosity is no longer required; see https://bugs.launchpad.net/snapcraft/+bug/1980515